### PR TITLE
added allow text input to date time picker

### PR DESCRIPTION
**Pull request by modern-dev-dude**
| Q               | A
| --------------- | ---
| Bug fix?        | [ ]
| New feature?    | [x ]
| New sample?      | [ ]
| Related issues?  | Adds enhancement  #928 , partially #Y, mentioned in #Z

#### What's in this Pull Request?

set allowTextInput to true
added handleTextChange for specific error message when entering a string
added loc string for english textErrorMessage 
added DateTimePickerTextErrorMessage to the IDateTimePickerInterface
added missing dependency for unit testing used in setup.js
added init test for date time picker to mount, render, update, and unmount

### CODE REVIEW

File: src/controls/dateTimePicker/DateTimePickerStrings.ts [hunk 1]
```diff
@@ -95,4 +95,10 @@ export class DateTimePickerStrings implements IDateTimePickerStrings {
    * Separator between time of day components (hours, minutes, seconds).
    */
   public timeSeparator: string = strings.DateTimePickerTimeSeparator;
+  /**
+   * error message when text is entered in the date picker
+   *  ?? added for a default label until loc strings are completed
+   */
+   public textErrorMessage: string = strings?.DateTimePickerTextErrorMessage ?? "Please enter a valid date";
```
Comment (joelfmrodrigues) [inline src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: @modern-dev-dude can you please use a localised string if DateTimePickerTextErrorMessage is not defined?

Comment (joelfmrodrigues) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: Actually, when would strings not be defined?

Comment (modern-dev-dude) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: @joelfmrodrigues 
Correct me if I am wrong and I will go update the PR, but it appeared to be the localized strings are manually entered for several different languages.  
So I added the English one in the file loc.en on line 223:
DateTimePickerTextErrorMessage:"Please enter a valid date"

I am assuming if the developer uses other locals than English DateTimePickerTextErrorMessage will be undefined. Unless those are handled elsewhere. 

If you have any recommendations for converting English to the other Locals, I'd be happy to insert them all and remove the undefined check

Comment (AJIXuMuK) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: If the key is missing in the translation, we should add it, even with the English translation in it.
In the code of the controls we need to assume that the strings are defined.

Comment (AJIXuMuK) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: If it's a new localization key, we can run Azure translation services that will create the needed key in all translations files.

Comment (modern-dev-dude) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: is this a service I can run myself? I was trying to take the work off you guys haha

Comment (modern-dev-dude) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: And for clarity, The next action item for me is to add the key to all the loc files with the English translation?

Comment (AJIXuMuK) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: If you have an Azure AD subscription - you can potentially run it yourself :)
Just leave the key for English, and I will run the script after the PR is merged.

Comment (modern-dev-dude) [reply src/controls/dateTimePicker/DateTimePickerStrings.ts:102]: Got it. Thanks.

**Comment by joelfmrodrigues** [comment]
@modern-dev-dude did you just include a lot of changes by accident? Was that a pull from dev or something like that?

**Comment by modern-dev-dude** [comment]
> @modern-dev-dude did you just include a lot of changes by accident? Was that a pull from dev or something like that?

I believe I pulled from dev before removing the nullish coalescing operator commit. If I need to rebase let me know.

**Comment by joelfmrodrigues** [comment]
@modern-dev-dude could you please do that? The PR is now displaying a large number of changes

**Comment by modern-dev-dude** [comment]
@joelfmrodrigues 
I opened a new PR removing all the extra commits  
Closing this pr now so there is no dups
https://github.com/pnp/sp-dev-fx-controls-react/pull/971

**Pull request closed without merging**